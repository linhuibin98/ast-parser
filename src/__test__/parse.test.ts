import { describe, expect, test } from 'vitest'
import { parse } from '..'

describe('testTokenizerFunction', () => {
  test('example', () => {
    const code = 'let a = function() {}'
    expect(parse(code)).toMatchSnapshot()
  })
})
