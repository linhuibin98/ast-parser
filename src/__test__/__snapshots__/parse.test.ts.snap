// Vitest Snapshot v1

exports[`testTokenizerFunction > example 1`] = `
{
  "body": [
    {
      "declarations": [
        {
          "end": 21,
          "id": {
            "end": 5,
            "name": "a",
            "start": 4,
            "type": "Identifier",
          },
          "init": {
            "body": {
              "body": [],
              "end": 21,
              "start": 19,
              "type": "BlockStatement",
            },
            "end": 21,
            "id": null,
            "params": [],
            "start": 8,
            "type": "FunctionExpression",
          },
          "start": 4,
          "type": "VariableDeclarator",
        },
      ],
      "end": 21,
      "kind": "let",
      "start": 0,
      "type": "VariableDeclaration",
    },
  ],
  "end": 21,
  "start": 0,
  "type": "Program",
}
`;
